### PR TITLE
V2.0.0 - Loss function with penalty

### PR DESCRIPTION
**Pull request by kovanostra**
- Obtains data from postgresql table
- Obtains a penalty matrix from a postgresql table
- Uses loss function with penalty based on the feasibility of phi, psi angles of the labels
- Various other refactoring

### CODE REVIEW

File: PARAMETERS.md [hunk 1]
```diff
@@ -22,48 +22,48 @@ DEVICE='cpu'
 
 **USED FOR GRID SEARCH**
```
Comment (mathisi-ai) [inline PARAMETERS.md:23]: Comments, really???

File: README.md [hunk 2]
```diff
@@ -110,13 +114,15 @@ This repository contains two dataset folders with examples of data to run the co
...
-### 9. Environment variables
+### 9. Grid search and inference parameters
       
-The model and grid search can be set up using a set of environment variables contained in the grid-search-parameters.sh. Please refer to the ENVIRONMENT_VARIABLES.md for the full list of available environment variables and how to use them.
+The grid search and inference parameters can be set up using a parameter json file contained in the parameters directory. Please refer to the PARAMETERS.md for the full list of available parameters and how to use them.
 
 ### 10. Execute a grid search
        
-Before executing a grid-search please go to the grid-search.sh to add your PYTHONPATH=path/to/message-passing-nn/.
+Before executing a grid-search please go to the grid-search.sh to add your:
```
Comment (mathisi-ai) [inline README.md:123]: I feel the need for a web application for set-up........

File: message_passing_nn/fixtures/amino_acid_maps.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+map_amino_acid_codes = {
```
Comment (mathisi-ai) [inline message_passing_nn/fixtures/amino_acid_maps.py:1]: Would still prefer the standard list but ok

File: tests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+from message_passing_nn.utils.postgres_connector import PostgresConnector
+
+os.environ['DATASET_TABLE'] = TEST_DATASET
+os.environ['PENALTY_TABLE'] = TEST_PENALTY
+os.environ["DATABASE"] = PDB_TEST
+os.environ["POSTGRES_USERNAME"] = POSTGRES
+os.environ["POSTGRES_PASSWORD"] = POSTGRES
+os.environ["POSTGRES_HOST"] = LOCALHOST
+os.environ["POSTGRES_PORT"] = PORT
+
```
Comment (mathisi-ai) [inline tests/__init__.py:13]: Should this not be in a config file

File: tests/fixtures/environment_variables.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+LOCALHOST = "localhost"
+
+POSTGRES = "postgres"
+
+PDB_TEST = "pdb_test"
+
+TEST_DATASET = "message_passing_tests"
+
+TEST_PENALTY = "message_passing_penalty_tests"
```
Comment (mathisi-ai) [inline tests/fixtures/environment_variables.py:11]: There seems to be lots of different files that contain parameters of some sort

File: tests/graph/__init__.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import os
+
+from tests.fixtures.environment_variables import *
+
+os.environ['TABLE'] = TEST_DATASET
+os.environ["DATABASE"] = PDB_TEST
+os.environ["POSTGRES_USERNAME"] = POSTGRES
+os.environ["POSTGRES_PASSWORD"] = POSTGRES
+os.environ["POSTGRES_HOST"] = LOCALHOST
+os.environ["POSTGRES_PORT"] = PORT
```
Comment (mathisi-ai) [inline tests/graph/__init__.py:10]: These already exist in a file, no?

File: tests/use_case/__init__.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import os
+
+from tests.fixtures.environment_variables import *
+
+os.environ['TABLE'] = TEST_DATASET
+os.environ["DATABASE"] = PDB_TEST
+os.environ["POSTGRES_USERNAME"] = POSTGRES
+os.environ["POSTGRES_PASSWORD"] = POSTGRES
+os.environ["POSTGRES_HOST"] = LOCALHOST
+os.environ["POSTGRES_PORT"] = PORT
```
Comment (mathisi-ai) [inline tests/use_case/__init__.py:10]: Same comment on parameters in may different files

File: tests/use_case/test_grid_search.py [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+            "epochs": [10, 15],
+            "batch_size": [3, 4],
+            "validation_split": [0.2],
+            "test_split": [0.1],
+            "loss_function": ["MSE"],
+            "optimizer": ["Adagrad"],
+            "time_steps": [1],
+            "validation_period": [5],
+            "scaling_factor": [0.1],
+            "penalty_decimals": [0]
```
Comment (mathisi-ai) [inline tests/use_case/test_grid_search.py:128]: Fixed values? Dude?

**Review by mathisi-ai — CHANGES_REQUESTED** [review]
Dude, you are a STALLION

File: PARAMETERS.md (hunk 1 above)
Comment (kovanostra) [reply PARAMETERS.md:23]: Second place you are allowed to have comments is the documentation haha

File: README.md (hunk 2 above)
Comment (kovanostra) [reply README.md:123]: Get your ass to work! (After we merge what you already have)

File: tests/__init__.py (hunk 4 above)
Comment (kovanostra) [reply tests/__init__.py:13]: No, it's in the tests. Not really a need for that

File: tests/graph/__init__.py (hunk 6 above)
Comment (kovanostra) [reply tests/graph/__init__.py:10]: Again it's tests

File: tests/use_case/test_grid_search.py (hunk 8 above)
Comment (kovanostra) [reply tests/use_case/test_grid_search.py:128]: Man check the main code, not the tests. The tests are supposed to have fixed values

**Review by mathisi-ai — APPROVED** [review]

**Comment by mathisi-ai** [comment]
GET YOUR FUCKING AMINO ACIDS IN THE RIGHT ORDER

File: tests/use_case/__init__.py (hunk 7 above)
Comment (kovanostra) [reply tests/use_case/__init__.py:10]: It is so that they can be run independently.

**Pull request merged**